### ISSUE TITLE
Set custom `JobName`

### ISSUE THREAD

**Issue by ElPicador** [spotify/scio#338]
When specifying a jobName from command line the jobName seems to be is overridden by scio:

To reproduce:
```
> run-main YourJob --runner=DataflowPipelineRunner --streaming --update --jobName=JOB_NAME --etc.
[info] Running YourJob --runner=DataflowPipelineRunner --streaming ...
Dataflow SDK version: 1.8.0
[error] (run-main-1) java.lang.IllegalArgumentException: Could not find running job named topks-rcschermesser-1115160752
java.lang.IllegalArgumentException: Could not find running job named topks-rcschermesser-1115160752
	at com.google.cloud.dataflow.sdk.runners.DataflowPipelineRunner.getJobIdFromName(DataflowPipelineRunner.java:3210)
	at com.google.cloud.dataflow.sdk.runners.DataflowPipelineRunner.run(DataflowPipelineRunner.java:613)
	at com.google.cloud.dataflow.sdk.runners.DataflowPipelineRunner.run(DataflowPipelineRunner.java:200)
	at com.google.cloud.dataflow.sdk.Pipeline.run(Pipeline.java:181)
	at com.spotify.scio.ScioContext$$anonfun$close$1.apply(ScioContext.scala:303)
	at com.spotify.scio.ScioContext$$anonfun$close$1.apply(ScioContext.scala:297)
	at com.spotify.scio.ScioContext.requireNotClosed(ScioContext.scala:343)
	at com.spotify.scio.ScioContext.close(ScioContext.scala:297)
	at algolia.harrylogger.runners.log.TopKs$.main(TopKs.scala:42)
	at algolia.harrylogger.runners.log.TopKs.main(TopKs.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
```

I suspect those lines to be the issue: https://github.com/spotify/scio/blob/master/scio-core/src/main/scala/com/spotify/scio/ScioContext.scala#L293-L294

**Comment by ElPicador** [comment]
I understand the need to have automatic naming for BatchJobs. 
So perhaps an easy fix would be to see if we are on a streaming job and do not set the name if so. WDYT?

**Comment by andrewsmartin** [comment]
@ElPicador were you trying to set the name for a new job or update an existing job?

**Comment by ElPicador** [comment]
I tried both.
My exemple is indeed for an update

**Comment by nevillelyh** [comment]
I think the currently ScioContext sets app name and job name is automatically derived from it. Should be easy to fix to allow override. Mind submitting a PR?

**Comment by andrewsmartin** [comment]
@ElPicador did you want to submit a PR for this? If not we can fix.

**Comment by ElPicador** [comment]
Yes will do. I'll do it next week.

**Comment by andrewsmartin** [comment]
@ElPicador Actually, it turns out some of our internal users are affected by this issue and we would like to get the fix out today. I will take care of this.

**Comment by ravwojdyla** [comment]
Closed via #341
